storage/innobase/dict/dict0dd.cc

```diff
@@ -6220,22 +6224,17 @@ static bool dd_update_tablespace_dd_flags(
     waited_so_far_for_lock += lock_wait_timeout;
   }
 
-  waited_so_far_for_lock = 0;
-  uint32_t dd_space_flags = 0;
+  if (dd_space == nullptr) {
+    dd::commit_or_rollback_tablespace_change(thd, dd_space, true);
+    return true;
+  }
 
-  if (dd_space == nullptr ||
-      dd_space->se_private_data().get(dd_space_key_strings[DD_SPACE_FLAGS],
-                                      &dd_space_flags)) {
+  if (update(dd_space)) {
     dd::commit_or_rollback_tablespace_change(thd, dd_space, true);
     return (true);
   }
```

Comment:
[optional] Can we combine these two?

Reply:
On recovery all the tables that have some redo logs to apply - have theirs page0 validated. So if there is in progress online encryption and there is no DD flag for it - the page0 will be read anyways. After it is read - crypt_data will be read and all the keys that are used by the tablespace will be checked.

Reply:
I mean, combining these steps would still be the logical equivalent before combining.

Reply:
yes sure, I do not know what I was referring to in my previous comment.